src/Text/Markdown/SlamDown.purs

```diff
@@ -197,14 +199,53 @@ everythingM b i (SlamDown bs) = mconcat <$> traverse b' bs
 everything :: forall r. (Monoid r) => (Block -> r) -> (Inline -> r) -> SlamDown -> r
 everything b i = runIdentity <<< everythingM (pure <<< b) (pure <<< i)
 
-eval :: (Maybe String -> List String -> String) -> SlamDown -> SlamDown
-eval f = everywhere b i
+-- | Evaluates code blocks embedded within a SlamDown document.
+-- | - The first function handles the evaluation of fenced code blocks and
+-- |   inline code values. The first argument is the type associated with a
+-- |   fenced code block, taken from the opening triple-backtick. The second
+-- |   argument is each line of the code block. In the case of inline code the
+-- |   first argument is always `Nothing` and the `List` will only contain one
+-- |   item.
+-- | - The second function handles code blocks used to provide a default value
+-- |   for a text-based input.
+-- | - The third function handles singular values used in determining the
+-- |   selected value in a dropdown or collection of radio buttons.
+-- | - The fourth function handles lists of values used for the list of options
+-- |   in a dropdown, the list of values for checkboxes, and the list of
+-- |   selected values for checkboxes.
+eval :: forall m. (Monad m)
```

Comment:
Record makes sense, I'll do that.

The case where HTML (or non-string eval results) would perhaps be useful is for the `Code` and `CodeBlock` cases, however I'm not really sure how to get that in there without making `SlamDown` a functor so it can be typed with whatever type those values would evaluate to - as currently this library knows nothing about HTML, it goes through `purescript-markdown-halogen` afterwards to be rendered to that.

There may be some benefit in the functorified approach though, as then you'd know whether a piece of SlamDown has been evaluated or not, as it could flip from `SlamDown Void` to `SlamDown (HTML p i)` or something like that.

For the form items it does make sense, as that's currently the internal representation used for all form related values. They could be made more flexible, but we'd need some extra machinery by typeclass of whatever to provide a means of stringifying the values for dropdowns/checkboxes/radio buttons, and have some way of testing equality of the values (I guess that could just be as simple as `Show` and `Eq` though).